src/server/api/routers/home.ts

```diff
@@ -0,0 +1,236 @@
+import { TRPCError } from '@trpc/server';
+import { and, asc, eq } from 'drizzle-orm';
+import { useTranslationsFromContext } from '@/server/api/locale';
+import { protectedProcedure, publicProcedure } from '@/server/api/procedures';
+import { createRouter } from '@/server/api/trpc';
+import {
+  homeCarouselSlideLocalizations,
+  homeCarouselSlides,
+} from '@/server/db/tables';
+import { deleteFile, getFileUrl, insertFile } from '@/server/services/files';
+import { changeSlideActiveSchema } from '@/validations/home/changeSlideActiveSchema';
+import { editSlideSchema } from '@/validations/home/editSlideSchema';
+import { selectSlideSchema } from '@/validations/home/fetchSlideSchema';
+import { fetchSlidesSchema } from '@/validations/home/fetchSlidesSchema';
+import { slideSchema } from '@/validations/home/slideSchema';
+
+const homeRouter = createRouter({
+  fetchSlide: publicProcedure
+    .input((input) =>
+      selectSlideSchema(useTranslationsFromContext()).parse(input),
+    )
+    .query(async ({ ctx, input }) => {
+      const slide = await ctx.db.query.homeCarouselSlides.findFirst({
+        where: eq(homeCarouselSlides.id, input.id),
+        with: {
+          localizations: true,
+        },
+      });
+
+      if (!slide) {
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: ctx.t('home.api.slideNotFound'),
+          cause: { toast: 'error' },
+        });
+      }
+
+      return {
+        ...slide,
+        imageUrl: slide.imageId ? await getFileUrl(slide.imageId) : null,
+      };
+    }),
+  fetchSlides: publicProcedure
+    .input((input) =>
+      fetchSlidesSchema(useTranslationsFromContext()).parse(input),
+    )
+    .query(async ({ ctx, input }) => {
+      const slides = await ctx.db.query.homeCarouselSlides.findMany({
+        where: input.onlyActive
+          ? eq(homeCarouselSlides.active, true)
+          : undefined,
+        with: {
+          localizations: true,
+        },
+        orderBy: [asc(homeCarouselSlides.id)],
+      });
+
+      return Promise.all(
+        slides.map(async (slide) => {
+          return {
+            ...slide,
+            imageUrl: slide.imageId ? await getFileUrl(slide.imageId) : null,
+          };
+        }),
+      );
+    }),
+  createSlide: protectedProcedure
```

Comment:
Procedures related to editing/deleting slides should be limited to leadership and admins
```suggestion
  createSlide: leadershipProcedure
```